connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/ProtobufOptions.scala

```diff
@@ -38,6 +38,12 @@ private[sql] class ProtobufOptions(
 
   val parseMode: ParseMode =
     parameters.get("mode").map(ParseMode.fromString).getOrElse(FailFastMode)
+
+  // Setting the circularReferenceDepth to 0 allows the field to be recursed once, setting
+  // it to 1 allows it to be recursed twice, and setting it to 2 allows it to be recursed
+  // thrice. circularReferenceDepth value greater than 2 is not allowed. If the not
+  // specified, it will default to -1, which disables recursive fields.
+  val circularReferenceDepth: Int = parameters.getOrElse("circularReferenceDepth", "-1").toInt
```

Comment:
Suggestion for renaming this option:  _"recursive.fields.max.depth"_ 

_circularReferenceDepth_ sounds very code variable type.

Reply:
If we go with that, we could rename the variable as well to 'recursiveFieldMaxDepth' (but this is your choice).